plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployInfo.java

```diff
@@ -0,0 +1,65 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.osgi.framework.FrameworkUtil;
+import org.w3c.dom.DOMException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class AppEngineDeployInfo {
```

Comment:
So it seems to me that we should be able to specify a project ID and version explicitly (and shouldn't we support a service name here too?) so that a module/service can be republished to other projects without having to hardcode a name.

For example, think of an OEM that maintains separate AppEngine projects for clients that are built against a common codebase.